docs/api/ref/GetAgent.md

```diff
@@ -0,0 +1,253 @@
+
+```ts
+/** 
+ * @typedef {Object} GetAgentParams Type representing parameters passed to the 
+ * getAgent function. 
+ * 
+ * @property {string} identityUrl The app's current URL is normally sent to 
+ * a web-based desktop agent to help establish its identity. This property 
+ * may be used to override the URL sent (to handle situations where an app's 
+ * URL is not sufficiently stable to use for identity purposes). The URL set 
+ * MUST match the origin of the application (scheme, hostname, and port) or  
+ * it will be ignored. If not specified, the app's current URL will be used.  
+ * 
+ * @property {number} timeout Number of milliseconds to allow for an fdc3  
+ * implementation to be found before calling the failover function or 
+ * rejecting (default 1000). Note that the timeout is cancelled as soon as a 
+ * Desktop Agent is detected. There may be additional set-up steps to perform 
+ * which will happen outside the timeout. 
+ * 
+ * @property {boolean} channelSelector Flag indicating that the application  
+ * needs access to a channel selector UI (i.e. because it supports User Channels 
+ * and does not implement its own UI for selecting channels). If not set will 
+ * default to true. MAY be ignored by Desktop Agent Preload (container)  
+ * implementations. 
+ * 
+ * @property {boolean} intentResolver Flag indicating that the application  
+ * needs access to an intent resolver UI (i.e. because it supports raising on or 
+ * more intents and and does not implement its own UI for selecting target apps. 
+ * If not set will default to true. MAY be ignored by Desktop Agent Preload  
+ * (container) implementations. 
+ * 
+ * @property {boolean} dontSetWindowFdc3 For backwards compatibility, `getAgent` 
+ * will set a reference to the Desktop Agent implementation at `window.fdc3` 
+ * if one does not already exist, and will fire the fdc3Ready event. Setting  
+ * this flag to `false` will inhibit that behavior, leaving `window.fdc3` unset. 
+ *  
+ * @property {function} failover An optional function that provides a  
+ * means of connecting to or starting a Desktop Agent, which will be called 
+ * if no Desktop Agent is detected. Must return either a Desktop Agent  
+ * implementation directly (e.g. by using a proprietary adaptor) or a  
+ * WindowProxy (e.g a reference to another window returned by `window.open`  
+ * or an iframe's `contentWindow`) for a window or frame in which it has loaded 
+ * a Desktop Agent or suitable proxy to one that works with FDC3 Web Connection 
+ * and Desktop Agent Communication Protocols. 
+ */ 
+type GetAgentParams = { 
+    timeout?: number, 
+    identityUrl?: string, 
+    channelSelector?: boolean, 
+    intentResolver?: boolean,
+    dontSetWindowFdc3?: boolean,
+    failover?: (args: GetAgentParams) => Promise<WindowProxy | DesktopAgent> 
```

Comment:
weirdly, optional here but not in the preview... I guess that means https://github.com/finos/FDC3/issues/1319 is done ok yes it is.

Reply:
Indeed, I'll have to defer to you on the build @robmoffat - the commit that ficed this went in 8 mins *after* the last preview build. I don't know why it didn't trigger a new build in netlify.